### PR TITLE
mode must be integer

### PR DESCRIPTION
**Pull request by tavy315**
If the optional mode parameter is set to COUNT_RECURSIVE (or 1), count() will recursively count the array. This is particularly useful for counting all the elements of a multidimensional array.

### CODE REVIEW

**Comment by mikeSimonson** [comment]
@tavy315 Thanks, Can you update your PR to use COUNT_RECURSIVE instead, that would be more descriptive.

File: lib/Doctrine/DBAL/Migrations/Migration.php [hunk 1]
```diff
@@ -181,7 +181,7 @@ public function migrate($to = null, $dryRun = false, $timeAllQueries = false, ca
         $this->outputWriter->write("\n  <comment>------------------------</comment>\n");
         $this->outputWriter->write(sprintf("  <info>++</info> finished in %ss", $time));
         $this->outputWriter->write(sprintf("  <info>++</info> %s migrations executed", count($migrationsToExecute)));
-        $this->outputWriter->write(sprintf("  <info>++</info> %s sql queries", count($sql, 1) - count($sql)));
+        $this->outputWriter->write(sprintf("  <info>++</info> %s sql queries", count($sql, \COUNT_RECURSIVE) - count($sql)));
```
Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/Migration.php]: Please import it instead (`use const`) to respect our CS - which is still being applied in #590

**Comment by Majkl578** [comment]
@tavy315 Can you please squash the changes into single commit? Thanks.

**Comment by tavy315** [comment]
Sorry guys, I tried to squash, seems I'm doing it wrong. Can you please make sure it's done right?

**Comment by lcobucci** [comment]
@tavy315 it should be something like this (assuming that your fork remote name is `origin` and that our repo remote name is `upstream`):

```sh
$ git fetch -p upstream
$ git rebase -i upstream/master
```

This will open your editor with a list of your commits, like:

```
pick 92cb0c0 mode must be integer
pick 3b7641f Update Migration.php
pick c1f0f39 Update Migration.php
pick 939e5ca count mode must be integer

# Rebase 92cb0c0..939e5ca onto 2bdb138 (4 commands)
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
# d, drop = remove commit
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
```

You should then apply these changes (in your editor):

```diff
pick 92cb0c0 mode must be integer
pick 3b7641f Update Migration.php
pick c1f0f39 Update Migration.php
pick 939e5ca count mode must be integer

# Rebase 92cb0c0..939e5ca onto 2bdb138 (4 commands)
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
# d, drop = remove commit
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
```

**Comment by mikeSimonson** [comment]
@tavy315 Thanks

**Pull request merged**